ext/mysqli/mysqli_prop.c

```diff
@@ -71,38 +71,71 @@ if (!obj->ptr) { \
 	p = (MYSQL_STMT *)((MY_STMT *)((MYSQLI_RESOURCE *)(obj->ptr))->ptr)->stmt; \
 }
 
-#define MYSQLI_MAP_PROPERTY_FUNC_LONG( __func, __int_func, __get_type, __ret_type, __ret_type_sprint_mod)\
+#define MYSQLI_MAP_PROPERTY_FUNC_LONG_OR_STR( __func, __int_func, __get_type, __ret_type, __ret_type_sprint_mod)\
 static int __func(mysqli_object *obj, zval *retval, bool quiet) \
 {\
 	__ret_type l;\
 	__get_type;\
-	if (!p) {\
-		ZVAL_NULL(retval);\
-	} else {\
+	if (p) {\
 		l = (__ret_type)__int_func(p);\
 		if (l < ZEND_LONG_MAX) {\
```

Comment:
It's a bit of a shame, but I don't exactly know when overflow can happen. So currently I only check for overflow when a mysql function has a return type of `my_ulonglong`. If I remember correctly, `zend_long` is `long long` (?). Unfortunately I couldn't find its declaration now.

Reply:
Yes, I think only handling overflow for my_ulonglong is right. Others could theoretically overflow on 32-bit systems (unsigned vs signed), but can't in practice -- you're not going to get a table with more than 2 billion columns :)